src/olympia/api/middleware.py

```diff
@@ -19,3 +19,26 @@ def process_response(self, request, response):
 
     def process_exception(self, request, exception):
         self.identify_request(request)
+
+
+class APICacheControlMiddleware:
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        response = self.get_response(request)
+        request_conditions = (
+            request.is_api
+            and request.method in ('GET', 'HEAD')
+            and 'HTTP_AUTHORIZATION' not in request.META
+            and 'disable_caching' not in request.GET
+        )
+        response_conditions = (
+            not response.cookies
```

Comment:
do we set cookies for API calls?

Reply:
Yes, in the authentication flow (and logout, but that's a `DELETE`)